src/buckets/components/context/lineProtocol.tsx

```diff
@@ -0,0 +1,137 @@
+// Libraries
+import React, {FC, useCallback, useState} from 'react'
+import {useSelector} from 'react-redux'
+
+// Utils
+import {postWrite} from 'src/client'
+import {getErrorMessage} from 'src/utils/api'
+
+// Selectors
+import {getOrg} from 'src/organizations/selectors'
+
+// Types
+import {RemoteDataState, LineProtocolTab, WritePrecision} from 'src/types'
+
+export type Props = {
+  children: JSX.Element
+}
+
+export interface LineProtocolContextType {
+  body: string
+  handleResetLineProtocol: () => void
+  handleSetBody: (_: string) => void
+  handleSetTab: (tab: LineProtocolTab) => void
+  precision: WritePrecision
+  setPrecision: (_: WritePrecision) => void
+  tab: LineProtocolTab
+  writeError: string
+  writeLineProtocol: (_: string) => void
+  writeStatus: RemoteDataState
+}
+
+export const DEFAULT_CONTEXT: LineProtocolContextType = {
+  body: '',
+  handleResetLineProtocol: () => {},
+  handleSetBody: (_: string) => {},
+  handleSetTab: (_: LineProtocolTab) => {},
+  precision: WritePrecision.Ns,
+  setPrecision: (_: WritePrecision) => {},
+  tab: 'Upload File',
+  writeError: '',
+  writeLineProtocol: (_: string) => {},
+  writeStatus: RemoteDataState.NotStarted,
+}
+
+export const LineProtocolContext = React.createContext<LineProtocolContextType>(
+  DEFAULT_CONTEXT
+)
+
+export const LineProtocolProvider: FC<Props> = React.memo(({children}) => {
+  const [body, setBody] = useState('')
+  const [tab, setTab] = useState<LineProtocolTab>('Upload File')
+  const [precision, setPrecision] = useState(WritePrecision.Ns)
+  const [writeStatus, setWriteStatus] = useState(RemoteDataState.NotStarted)
+  const [writeError, setWriteError] = useState('')
+
+  const org = useSelector(getOrg).name
+
+  const handleResetLineProtocol = () => {
+    setBody('')
+    setWriteStatus(RemoteDataState.NotStarted)
+    setWriteError('')
+  }
+
+  const writeLineProtocol = useCallback(
+    async (bucket: string) => {
+      try {
```

Comment:
For context, Subir and I offlined a discussion as to the implementation details of having a try / catch with an error being thrown at the end of it all. My main argument was simply that this was copy pasta'd from the original implementation, and since this is a refactor for usage sack, i didn't want to mess around too much with the implementation. Having said that, it might be valuable for us to revisit this at a later stage if it becomes problematic